File: javascript/leetcode/1299. Replace Elements with Greatest Element on Right Side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function (arr) {
  const sz = arr.length - 1;

  let max = arr[sz];

  for (let i = sz - 1; i >= 0; --i) {
    let val = arr[i];
    arr[i] = max;
    max = Math.max(val, max);
  }
  arr[sz] = -1;
  return arr;
};

File: c++/leetcode/1299. Replace Elements with Greatest Element on Right Side.cpp
class Solution
{
public:
  vector<int> replaceElements(vector<int> &arr)
  {
    const int sz = arr.size() - 1;

    int mx = arr[sz];

    for (int i = sz - 1; i >= 0; --i)
    {
      auto val = arr.at(i);
      arr[i] = mx;
      mx = max(val, mx);
    }
    arr[sz] = -1;

    return arr;
  }
};